routes/signup.js

```diff
@@ -69,75 +67,62 @@ router.post('/register', function(req, res) {
                   message: "We sent you an email for email verification. Please confirm your email."
               });
 
-              let userID;
-
-              //query the new inserted user to get the user-id and email verification code
-              connection.query(
-                'SELECT * FROM users WHERE email = ?',
-                [req.body.email],
-                function(error, result, fields) {
-                  if (error) throw error;
-                  userID = result.id;
-                   //use sendgrid to send email
-                  const email_verification = {
-                    to: req.body.email,
-                    from: 'simon@acceptmycrypto.com',
-                    subject: 'Please click the link below to verify your email!',
-                    text: 'Thank You for signing up! Go to this url to complete the registration.',
-                    html: `<a href="http://localhost:3001/email-verify/${userID}/${result.email_verification_token}>Verify My Email</a>`
-                  };
-                  sgMail.send(email_verification);
-                }
-              );
-
-              //Once the user clicks on the email verification, we get the id and email verification params
-              router.get('/email-verify/:user_id/:email_verification_token', function(req, res) {
-                console.log("params: ", req.params);
+              let UserID; //make a available to be resused when insert into users_cryptos table
+
+                //query the new inserted user to get the user-id and email verification code
                 connection.query(
-                  'SELECT * FROM users WHERE id = ?',
-                  [req.params.user_id],
+                  'SELECT * FROM users WHERE email = ?',
+                  [req.body.email],
                   function(error, result, fields) {
                     if (error) throw error;
+                    userID = result[0].id;
+
+                    //use sendgrid to send email
+                    let link = "http://localhost:3001/email-verify/" + userID + "/" + result[0].email_verification_token;
+
+                    const email_verification = {
+                      to: req.body.email,
+                      from: 'simon@acceptmycrypto.com',
+                      subject: 'Please click the link below to verify your email.',
+                      html: `<a href=${link}>Verify Email</a>`
+                    };
+                    sgMail.send(email_verification);
+                  }
+                );
+
+                //insert selected cryptos into users_cryptos table
+                connection.query(
+                  "SELECT * FROM crypto_metadata LEFT JOIN crypto_info ON crypto_metadata.crypto_name = crypto_info.crypto_metadata_name WHERE crypto_name IN (?)",
```

Comment:
Could you Select crypto_info.id or crypto_metadata id instead of * that way selectedCryptos will be an array of all the id's. Then you might not need to do lines 101 to 103.